src/libraries/Microsoft.Extensions.Hosting.Systemd/src/SystemdHelpers.cs

```diff
@@ -16,21 +16,24 @@ namespace Microsoft.Extensions.Hosting.Systemd
     /// </summary>
     public static partial class SystemdHelpers
     {
-        private static bool? _isSystemdService;
+        private static readonly bool _isSystemdService = GetIsSystemdService();
 
         /// <summary>
         /// Check if the current process is hosted as a systemd Service.
         /// </summary>
         /// <returns>
         /// <see langword="true" /> if the current process is hosted as a systemd Service; otherwise, <see langword="false" />.
         /// </returns>
-        public static bool IsSystemdService()
-            => _isSystemdService ??= GetIsSystemdService();
+        public static bool IsSystemdService() => _isSystemdService;
 
         private static bool GetIsSystemdService()
         {
             // No point in testing anything unless it's Unix
+#if NETCOREAPP
+            if (OperatingSystem.IsLinux())
+#else
```

Comment:
Is systemd ever supported on other Unix? Should this method add a `SupportedOSPlatformGuard`?

Reply:
- https://systemd.io/
- https://en.wikipedia.org/wiki/Systemd

From the introduction, think it's for Linux only, but not for sure

Reply:
reverted